Explaining autocast in release post (#225)
crystal-lang/crystal-website: _releases/2022-01-06-1.3.0-released.md

## Number Autocast

Primitive numeric values are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32` (note that before 1.3.0 only number _literals_ were autocasted):

<div class="code_section">{% highlight crystal %}
def foo(x : Int64)
  x
end

> foo 1_32 # Works in 1.2.2

Reviewer: ```suggestion
foo 1_32 # Works in 1.2.2 and 1.3.0
```

## Number Autocast

> Numeric elements are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32`:

Reviewer: ```suggestion
Primitive numeric values are now _autocasted_ to fit into larger types. For instance, it is possible to call a function expecting an `Int64` with an `Int32`:
```

<div class="code_section">{% highlight crystal %}
def foo(x : Int64)
  x
end

> def bar
>   1_i32
> end
> 
> foo(bar)

Reviewer: I don't think there's a need to have a method here?

```suggestion
bar = 1_i32
foo(bar)
```

{% endhighlight %}</div>

> Similarly, unsigned integer types can be casted into larger signed ones, and it also works for `Float32` (to be casted as `Float64`).

Reviewer: ```suggestion
Unsigned integer types can be autocasted into larger signed ones. And autocasting also works for floating point types (`Float32` to `Float64`).
```

If there is ambiguity, for instance, because there are more than one option, the compiler throws an error:

<div class="code_section">{% highlight crystal %}
def foo(x : Int64)
  x
end

def foo(x : Int128)
  x
end

> def bar
>   1_i32
> end

Reviewer: ```suggestion
bar = 1_i32
```
